sdk/identity/azure-identity/src/managed_identity_credential.cpp

```diff
@@ -11,13 +11,15 @@ namespace {
 std::unique_ptr<_detail::ManagedIdentitySource> CreateManagedIdentitySource(
     std::string const& credentialName,
     std::string const& clientId,
+    std::string const& resourceId,
```

Comment:
nit: Do you want to keep passing resourceId as `ResourceIdentifier` and not string here, so that in the future it is easier to modify? Because if we end up adding something there, it is likely because one of the ManagedIdentitySources knows how to use that extra information, so next thing we'll likely be doing is modifying these lines that pass a resourceId as a string to also carry that new piece of information, so why not just do it now - pass it as `ResourceIdentifier`, and where you use it as string, call `ToString()`?

Reply:
Interesting question. Atm, I don't see a reason to use anything from the `ResourceIdentifier` other the raw string form within the credential, in the foreseeable future.

Given how other languages use this type in identity (just as a string to pass in the query parameter), I also don't see us adding any more "components" to this type (only validation and parsing helpers which identity won't use).

In the off chance we need it, I am OK passing string here until the need to pass `ResourceIdentifier` arises.

Reply:
Hmm, based on what you said, are you sure we need this type at this moment then? Can't we just take std::stirng in the ManagedIdentityCred constructor instead?

Reply:
We can't take `std::string` (and need this string wrapper) due to ctor overloading.
We already have a ctor with `std::string`, which is `clientId`. We can't create another one with `std::string` resourceId.